Solutions/Corvus.UriTemplates/Corvus.UriTemplates/TemplateParameterProviders/VariableSpecification.cs

```diff
@@ -110,7 +108,8 @@ static void TryFormat(int value, Span<char> span, int digits)
     /// <returns>The variable specification as a string.</returns>
     public override string ToString()
     {
-        StringBuilder builder = StringBuilderPool.Shared.Get();
+        Span<char> initialBuffer = stackalloc char[3 + this.VarName.Length + this.PrefixLength];
```

Comment:
Should we check that `VarName` and `PrefixLength` both have sane lengths, and fall back to heap allocation if this turns out to be too long?

Reply:
Yes - I don't like stackalloc without the arraypool fallback if it is isn't a known fixed size.

Reply:
We decided just to throw if it's over a 4K threshold, because that's almost certainly indicative of a problem. (It suggests the whole URI is going to be even longer, and 4K is typically too long.)